include/nvexec/stream/let_xxx.cuh

```diff
@@ -110,17 +110,17 @@ namespace nvexec::STDEXEC_STREAM_DETAIL_NS {
           using op_state_t = __minvoke<__op_state_for<_Receiver, _Fun>, _As...>;
 
           cudaStream_t stream = __self.__op_state_->get_stream();
-
           result_sender_t* result_sender = static_cast<result_sender_t*>(
             __self.__op_state_->temp_storage_);
           kernel_with_result<_As&&...><<<1, 1, 0, stream>>>(
             std::move(__self.__op_state_->__fun_), result_sender, (_As&&) __as...);
 
           if (cudaError_t status = STDEXEC_DBG_ERR(cudaStreamSynchronize(stream));
               status == cudaSuccess) {
+            __self.__op_state_->defer_temp_storage_destruction(result_sender);
```

Comment:
If `status != cudaSuccess`, then the result sender may or may not be constructed, right? How can we know for certain whether the destructor needs to be called?

Reply:
The destructor is registered only when `status == cudaSuccess`, which should be an improvement compared to the current state. If the `status != cudaSuccess`, it's still possible that the constructor was called and we miss the destruction, but I suggest we address this as a follow up PR (once the mechanism is in place).